cpp/src/gandiva/cache_test.cc

```diff
@@ -38,5 +42,77 @@ TEST(TestCache, TestGetPut) {
   ASSERT_EQ(cache.GetObjectCode(TestCacheKey(2)), "world");
 }
 
-TEST(TestCache, TestGetCacheCapacity) { ASSERT_EQ(GetCapacity(), 5000); }
+namespace {
+constexpr auto capacity_env_var = "GANDIVA_CACHE_SIZE";
+constexpr auto default_capacity = 5000;
+}  // namespace
+
+TEST(TestCache, TestGetCacheCapacityDefault) {
+  ASSERT_EQ(GetCapacity(), default_capacity);
+}
+
+TEST(TestCache, TestGetCacheCapacityEnvVar) {
+  // Uncleared env var may have side-effect to subsequent tests. Use a structure to help
+  // clearing the env var when leaving the scope.
+  struct ScopedEnvVar {
```

Comment:
Can use the existing `EnvVarGuard` from our testing utilities.

Reply:
Ah, that's great to know. Will do. Thank you!